### PR TITLE
feat: Part 1 of ability to spin up different pipelines

### PR DESCRIPTION
**Pull request by tiina303**
## Problem

Longer term vision is PIPELINES also contains sth like 
`getInstance: async (hub: PipelineContext) => new NoopPipelineEngine(hub),` to get the pipeline, but for now, lets start with being able to define the topic and consumer group.

For hobby deploy & local dev (i.e. PluginServerMode = null)  we run all of the events ingestion pipelines.
In prod we can define to run each pipeline on a different host. We could run multiple on the same host, but for simplicity let's not do that now.

The pipelines by default don't have any ordering guarantees as by default new products aren't supposed to update persons (which isn't currently enfoced, but will be in follow-up PRs).

Made some tests a lot faster by only using necessary capabilities, which should lead to less flaky test failures for everyone going forward.
before:
Screenshot 2024-07-16 at 20 07 28
after:
Screenshot 2024-07-16 at 20 07 21

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## How did you test this code?

Added functional test for ingestion warnings. Verified in local kafka-ui that the message was sent to client_iwarnings_ingestion topic as we have them set-up in prod.

Locally made historical topic from python capture be client_iwarnings_ingestion and saw events being ingested normally.

### CODE REVIEW

**Review by oliverb123 — COMMENTED** [review]
The only major change here is that the topic and consumer group are passed to the constructor, rather than being pulled from global constants, is that fair to say?

Seems reasonable as implemented here to me.

**Comment by tiina303** [comment]
> The only major change here is that the topic and consumer group are passed to the constructor, rather than being pulled from global constants, is that fair to say?

Correct, I'm trying to make way to break things up, so progress can be made in multiple fronts at the same time. Specifically batching side & and runner execution side split.

**Review by xvello — APPROVED** [review]

File: plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export enum IngestionOverflowMode {
     RerouteRandomly, // discards partition locality
     ConsumeSplitByDistinctId,
     ConsumeSplitEvenly,
+    ConsumeSplitEventlyWithoutIngestionWarning,
```
Comment (oliverb123) [inline plugin-server/src/main/ingestion-queues/batch-processing/each-batch-ingestion.ts:36]: This isn't used by anything, right? It's identical to `ConsumeSplitEvenly`.

File: plugin-server/src/main/ingestion-queues/events-ingestion-consumer.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { status } from '../../utils/status'
+import { eachBatchParallelIngestion, IngestionOverflowMode } from './batch-processing/each-batch-ingestion'
+import { IngestionConsumer } from './kafka-queue'
+
+export type PipelineType = {
+    topic: string
+    consumer_group: string
+}
+
+export const PIPELINES: { [key: string]: PipelineType } = {
```
Comment (oliverb123) [inline plugin-server/src/main/ingestion-queues/events-ingestion-consumer.ts:15]: Can we use the constants in `kafka-topics.ts` here?

Comment (tiina303) [reply plugin-server/src/main/ingestion-queues/events-ingestion-consumer.ts:15]: the tricky thing is I don't want the person who creates the pipeline to need to touch multiple files

**Review by xvello — APPROVED** [review]

**Pull request merged**